ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt

```diff
@@ -49,56 +50,72 @@ class CoreAuthnRequestProtocolVerifier(
     /** 3.4 Authentication Request Protocol **/
     override fun verify() {
         super.verify()
+        CORE_3_4.start()
         verifyAuthnRequestProtocolResponse()
         verifySubjects()
         nameIdPolicyVerifier?.verify()
     }
 
+    /**
+     * 3.4.1 Element <AuthnRequest>
+     * Verify the Assertion Consumer Service URL
+     */
     fun verifyAssertionConsumerService(httpResponse: Response) {
+        CORE_3_4.start()
         val expectedACS = currentSPEntityInfo.getAssertionConsumerService(authnRequest, null,
                 authnRequest.assertionConsumerServiceIndex).url
         val actualACS = httpResponse.getLocation()
 
-        if (actualACS == null || actualACS != expectedACS)
-            throw SAMLComplianceException.create(SAMLCore_3_4_1_a,
+        if (actualACS == null || actualACS != expectedACS) {
+            Report.addExceptionMessage(SAMLComplianceException.create(SAMLCore_3_4_1_a,
                     message = "The URL at which the Response was received [$actualACS] does not" +
-                            " match the expected ACS URL [$expectedACS] based on the request.")
+                            " match the expected ACS URL [$expectedACS] based on the request."))
+        }
     }
 
+    /**
+     * 3.3.2.2.1 Element <RequestedAuthnContext>
+     * Verify responses of AuthnRequests With AuthnContext
+     */
     fun verifyAuthnContextClassRef() {
         samlResponse.recursiveChildren("AuthnContext")
                 .flatMap { it.children("AuthnContextClassRef") }
                 .firstOrNull { !ddfAuthnContextList.contains(it.textContent) }?.let {
-                    throw SAMLComplianceException.create(SAMLCore_3_3_2_2_1_a,
+                    Report.addExceptionMessage(SAMLComplianceException.create(SAMLCore_3_3_2_2_1_a,
                             message = """An <AuthnContextClassRef> that is not part of the
```

Comment:
✏️ Here's another usage of a raw string, outside the scope of your PR which is a little concerning.

Reply:
Did some searching and found a handful of other places where we shouldn't be using the raw string. I can fix this one  and those in a separate PR.